packages/contracts/src/yield/context/ITripleRateContext.sol

```diff
@@ -4,34 +4,35 @@ pragma solidity ^0.8.20;
 import { ICalcInterestMetadata } from "@credbull/yield/ICalcInterestMetadata.sol";
 
 /**
- * @title A triple rate context, with the 'full' rate and 2 reduced rates that apply temporally across 2 tenor periods.
- * @dev Context for yield calculations with a rate and dual reduced rates, applicable across Tenor Periods. All rate
- *  are expressed in percentage terms and scaled using [scale()]. The 'full' rate values are encapsulated by the
- *  [ICalcInterestMetadata].
+ * @title A triple rate context, with the 'full' interest rate and 2 reduced interest rates that apply temporally
+ *  across 2 tenor periods.
+ * @dev Context for yield calculations with an interest rate and dual reduced interest rates, applicable across Tenor
+ *  Periods. All rate are expressed in percentage terms and scaled using [scale()]. The 'full' rate values are
+ *  encapsulated by the [ICalcInterestMetadata].
  */
 interface ITripleRateContext is ICalcInterestMetadata {
+    /// @notice Associates an Interest Rate with the Tenor Period from which it applies.
+    struct TenorPeriodRate {
+        /// @dev The Interest Rate in percentage terms and scaled.
+        uint256 interestRate;
+        /// @dev The Tenor Period from which the associated rate applies.
+        uint256 effectiveFromTenorPeriod;
```

Comment:
hi @jplodge-pro - please clarify, what is effectiveFromTenorPeriod? 

with a tenor of 30, is it the following, or something else ?
- effectiveFromTenorPeriod0 : days 0-30
- effectiveFromTenorPeriod1 : days 31-60

Reply:
With your example, a tenor of 30, then when the `TripleRateContext` is created it sets the `_current` `TenorPeriodRate` to be `{ 0, 5.5% }`. Then the Operator updates the Reduced Rate (the T-bill rate, say 5.8%) setting on the 31st period and the `_current` becomes `{ 31, 5.8% }` and the `_previous` becomes `{ 0, 5.5% }`. 
The Operator could also set the rate on the 33rd period (naughty operator!), retroactively setting it to the 31st period.  
 So, yes, the `effectiveFromTenorPeriod` is the period from which that rate applies. I now suspect that our understandings are somehow orthogonal to each other! 
The term 'tenor period' I took from the existing code, I assumed you coined it to mean any period within a tenor?

Reply:
i think fromPeriod or effectiveFromPeriod is clearer. 

including the tenor in the name makes me wonder is this counting in 1 day or in 30 day increments.